### PR TITLE
adding regex_filter to cache spec with test case

### PR DESCRIPTION
**Pull request by gramhagen**
Addressing issue #289 

To support compressed files with extraneous files that should be ignored I'm adding an additional parameter to the cache spec.

```
sources:
  test:
    driver: csv
    args:
      urlpath: test.tar.gz
    cache:
      - type: compressed
        argkey: urlpath
        regex_filter: '.*sample.csv'
```

this is used to filter out the file(s) that are desired, one thing that doesn't seem ideal here is that the prefix will contain the full path to the cached files which were extracted, so you have to include wildcards for the beginning of the filter.

i created a test tarball with the calvert_uk data plus a readme file and added a test of this functionality

### CODE REVIEW

**Comment by martindurant** [comment]
This sounds very reasonable to me.
Am I right in thinking that all files are still extracted from the archive, but only the matching ones are returned to the driver for loading? That is a fine pattern (it can be useful to have any README locally), just wanted to be sure.

**Comment by gramhagen** [comment]
Yes, the files are extracted from the archive into the cache folder, but they are not cpatured in the cache metadata or visible to the driver.

**Comment by martindurant** [comment]
Cool, thanks for the clarification. Where do you think this parameter could be documented? Perhaps the docstring of `CompressedCache` should describe al the inputs.

**Comment by gramhagen** [comment]
yeah, that sounds good, i can update that. also perhaps the documentation can be updated, though I haven't look at that much. Are there are specific files / sections that you think would be helpful to change?

**Comment by martindurant** [comment]
The cache mechanism in general is mentioned in `catalog.rst`, under the YAML format section and says how to find the classes, so you can read their docstrings. The base cache class is in `api_base.rst`, but the actual implementations are not. It could have its own section there, perhaps.

**Comment by gramhagen** [comment]
Ok, I updated catalog.rst and added info in the docstring for the compressed cache class

File: docs/source/catalog.rst [hunk 1]
```diff
@@ -311,6 +311,33 @@ The "types" of caching are that supported are listed in ``intake.source.cache.re
 the docstrings of each for specific parameters that should appear in the cache block.
 
 
+Compressed Files
+----------------
```
Comment (martindurant) [inline docs/source/catalog.rst]: Sorry, I think this should be `'''''''''`, else it'll be a higher-level heading than intended

Comment (gramhagen) [reply docs/source/catalog.rst]: ok, no worries

**Comment by martindurant** [comment]
That'll do nicely, thank you very much.

**Pull request merged**